src/Tasks/dependency-check-build-task/dependency-check-build-task.ts

```diff
@@ -176,13 +209,41 @@ async function run() {
                 console.log(`##vso[build.uploadlog]${logFile}`);
         }
 
+        let message = "Dependency Check succeeded"
+        let result = tl.TaskResult.Succeeded
         if (failed) {
-            let message = "Dependency Check exited with an error code.";
-            if (isViolation) message = "CVSS threshold violation.";
+            if(isViolation) {
```

Comment:
Only fail upon found violations if "Only warn for found violations" option is false, if it was set (=true) then succeed-with-issues if a violation was found.